core/Azure.Mcp.Core/src/Logging/AzureEventSourceLogForwarder.cs

```diff
@@ -0,0 +1,153 @@
+    public AzureSdkEventSourceLogForwarder(ILoggerFactory loggerFactory, EventLevel level = EventLevel.Informational)
+    {
+        _loggerFactory = loggerFactory ?? throw new ArgumentNullException(nameof(loggerFactory));
+        _level = level;
+    }
+
+    /// <summary>
+    /// Called when a new EventSource is created. Enables listening to Azure-related event sources.
+    /// </summary>
+    /// <param name="eventSource">The event source that was created.</param>
+    protected override void OnEventSourceCreated(EventSource eventSource)
+    {
+        base.OnEventSourceCreated(eventSource);
+
+        // Listen to all Azure SDK event sources
+        // Azure SDK event sources start with "Azure-"
+        if (eventSource.Name?.StartsWith("Azure-", StringComparison.OrdinalIgnoreCase) == true)
+        {
+            try
+            {
+                EnableEvents(eventSource, _level, EventKeywords.All);
+
+                // Create a logger for this event source if not already created
+                if (!_loggers.ContainsKey(eventSource.Name))
+                {
+                    _loggers[eventSource.Name] = _loggerFactory.CreateLogger($"Azure.SDK.{eventSource.Name}");
+                }
+            }
+            catch (Exception)
+            {
+                // Ignore errors enabling event sources
+            }
+        }
+    }
+
+    /// <summary>
+    /// Called when an event is written. Forwards the event to the appropriate logger.
+    /// </summary>
+    /// <param name="eventData">The event data.</param>
+    protected override void OnEventWritten(EventWrittenEventArgs eventData)
+    {
+        if (eventData.EventSource?.Name == null)
+        {
+            return;
+        }
+
+        // Get or create a logger for this event source
+        if (!_loggers.TryGetValue(eventData.EventSource.Name, out var logger))
+        {
+            logger = _loggerFactory.CreateLogger($"Azure.SDK.{eventData.EventSource.Name}");
+            _loggers[eventData.EventSource.Name] = logger;
+        }
+
+        // Map EventSource EventLevel to Microsoft.Extensions.Logging LogLevel
+        var logLevel = MapEventLevel(eventData.Level);
+
+        // Format the message
+        var message = FormatMessage(eventData);
+
+        // Log the event
+        logger.Log(logLevel, eventData.EventId, message, null, (state, _) => state);
+    }
+
+    /// <summary>
+    /// Maps EventSource EventLevel to Microsoft.Extensions.Logging LogLevel.
+    /// </summary>
+    private static LogLevel MapEventLevel(EventLevel eventLevel) => eventLevel switch
+    {
+        EventLevel.Critical => LogLevel.Critical,
+        EventLevel.Error => LogLevel.Error,
+        EventLevel.Warning => LogLevel.Warning,
+        EventLevel.Informational => LogLevel.Information,
+        EventLevel.Verbose => LogLevel.Debug,
+        EventLevel.LogAlways => LogLevel.Information,
```

Comment:
Could you please explain a little bit on the level mapping? When mappin log level to event level, LogLevel.None is mapped to EventLevel.LogAlways but when mapping EventLevel.LogAlways to log level, it mapps to LogLevel.Information.

My naive understanding is that event level LogAlways should be Debug or Trace since it's even more Verbose than the Verbose event level.

Reply:
In systems such as ETW, LogAlways indicates "skip all level filtering".

In this case, we default to the Information level to prevent accidentally exposing sensitive information.